### PR TITLE
factor calculation for upscaled_texture from Crispy Doom

### PR DESCRIPTION
**Pull request by rfomin**
@ceski-1 please check if it works like Crispy.

### CODE REVIEW

**Comment by ceski-1** [comment]
Looks good:

Crispy (smooth pixel scaling: on):
crispy

Woof (enable bilinear sharpening: yes):
woof

File: src/i_video.c [hunk 1]
```diff
@@ -1423,6 +1423,67 @@ void I_GetScreenDimensions(void)
...
+    if (info.flags & SDL_RENDERER_SOFTWARE)
+    {
+        return;
+    }
+
+    // Get the size of the renderer output. The units this gives us will be
+    // real world pixels, which are not necessarily equivalent to the screen's
+    // window size (because of highdpi).
+
+    SDL_GetRendererOutputSize(renderer, &w, &h);
```
Comment (fabiangreffrath) [inline src/i_video.c:1442]: Will we need to call this on resize events?

Comment (rfomin) [reply src/i_video.c:1442]: Yes, it seem so. To simplify this, maybe we could use [`SDL_GetDesktopDisplayMode`](https://wiki.libsdl.org/SDL2/SDL_GetDesktopDisplayMode) here?

Comment (fabiangreffrath) [reply src/i_video.c:1442]: I meant on window resize events?

Comment (rfomin) [reply src/i_video.c:1442]: > I meant on window resize events?

Yes. I think we can create textures for maximum screen size and let SDL scale them for smaller window sizes.

Comment (fabiangreffrath) [reply src/i_video.c:1442]: Okay.

Comment (rfomin) [reply src/i_video.c]: Done in https://github.com/fabiangreffrath/woof/pull/1019/commits/c4bdc8dfe37ef3207c2b1f7f10bf78bc872797d1

**Review by fabiangreffrath — APPROVED** [review]
Thank you!

**Pull request merged**